### PR TITLE
Extract gpdb cluster start/stop from bdd library.

### PR DESCRIPTION
**Pull request by berlin-ab**
Allow each test to control the manipulation of the clusters.

### CODE REVIEW

**Comment by berlin-ab** [comment]
I do really like that.

> I think this will be helpful for the check tests that do not need to start cluster.
> 
> More curious what you think than a suggestion so don't let this block pushing - one pattern I see is that you always start cluster at beginning and stop and end of function. What would you think of wrapping that logic into its own callback that the scenario could use to decorate the functions? Sort of like:
> 
> ```c
> void
> withFiveCluster(void (*innerFunction)(void))
> {
>     startGpdbFiveCluster();
>     innerFunction();
>     stopGpdbFiveCluster();
> }
> 
> void
> withSixCluster(void (*innerFunction)(void))
> {
>     startGpdbSixCluster();
>     innerFunction();
>     stopGpdbSixCluster();
> }
> 
> void
> test_an_ao_table_with_data_can_be_upgraded(void **state)
> {
> 	given(withFiveCluster(createAoTableWithDataInFiveCluster));
> 	when(anAdministratorPerformsAnUpgrade);
> 	then(withSixCluster(aoTableShouldHaveDataUpgradedToSixCluster));
> }
> ```

I do really like that. Nice suggestion.

**Comment by berlin-ab** [comment]
note to self: The Makefile needs to be updated to match the name change of upgrade-bdd.c to bdd.c.

**Comment by berlin-ab** [comment]
@dgkimura Implemented and ready for feedback.

File: contrib/pg_upgrade/test/integration/utilities/bdd-helpers.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	current_step();
+	stopGpdbSixCluster();
+}
+
+bdd_step_function
+withinGpdbFiveCluster(bdd_step_function step)
+{
+	current_step = step;
+
+	return within_gpdb_five_function;
```
Comment (dgkimura) [inline contrib/pg_upgrade/test/integration/utilities/bdd-helpers.c]: Ah, if only we had lambdas..  Interestingly, seems GCC has an extension that could have solved this problem:

https://gcc.gnu.org/onlinedocs/gcc/Nested-Functions.html

Oh, well...

**Pull request merged**